samples/keys/hw_unique_key/README.rst

```diff
@@ -0,0 +1,82 @@
+.. _hw_unique_key_usage:
+
+Hardware Unique Key
+###################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This sample shows how to use a hardware unique keys (HUKs) to derive a key and use it for encryption, through the psa_crypto APIs.
+
+Overview
+********
+
+The sample does the following:
+
+* Initialize the HW if applicable
+  ** When TFM is enabled, it does this automatically on boot
+* Generate HUKs if applicable
+  ** When TFM is enabled, it does this automatically on boot
+* Generate a random IV
+* Derive a key
+  ** When TFM is enabled, this is done using the psa_crypto APIs
+  ** When TFM is not enabled, native APIs are used, and the key is imported into psa_crypto.
+* Encrypt a sample string using the key_id received from psa_crypto.
```

Comment:
```suggestion
The sample derives a key used to encrypt a string as follows:

1. Initializing the hardware, if applicable.

   When TF-M is enabled, the hardware is initialized automatically when booting.

#. Generating HUKs, if applicable.

   When TF-M is enabled, the HUKs are generated automatically when booting.
   
#. Generating a random initialization vector (IV).

#. Deriving a key used for encryption.

   When TF-M is enabled, the key is derived using the psa_crypto APIs.
   Otherwise, the native APIs are used, and the key is imported into the psa_crypto.

#. Encrypting a sample string using the key_id received from the psa_crypto.
```

Reply:
```suggestion
The sample goes through the following steps to derive a key and use it to encrypt a string:

1. Initializing the hardware, if applicable.

   When TF-M is enabled, the hardware is initialized automatically when booting.
   
#. Generating HUKs, if applicable.

   When TF-M is enabled, the HUKs are generated automatically when booting.
   
#. Generating a random initialization vector (IV).

#. Deriving a key used for encryption.

   When TF-M is enabled, the key is derived using the psa_crypto APIs.
   Otherwise, the native APIs are used, and the key is imported into psa_crypto.
   
#. Encrypting a sample string using the key_id received from the psa_crypto.
```